src/or/control.c

```diff
@@ -2207,6 +2207,25 @@ getinfo_helper_dir(control_connection_t *control_conn,
         return -1;
       }
     }
+  } else if (!strcmp(question, "md/all")) {
+    const smartlist_t *nodes = nodelist_get_all_nodes();
+    if (!nodes) {
```

Comment:
Hmm, I don't think we should return `Internal error` if there are no nodes/consensus. Instead we should do what we do for `ns/all` in `getinfo_helper_networkstatus()` and return an empty string.

That's also an artficat of the way your `nodelist_get_all_nodes()` function works. If you use the `nodelist_get_list()` function, you are not gonna have this problem.

Reply:
Fixed in 51641533b2c565e22e24b27bf5782d70802a051.